### PR TITLE
DAOS-6539 object: refresh DTX status during EC aggregation

### PR DESCRIPTION
**Pull request by Nasf-Fan**
It is possible that the EC aggregation may hit non-committed DTX
that be garbage DTX entry left by some test failure injection.
Under such case, the EC aggregation needs to call DTX refresh to
check with related DTX leader. Otherwise, the EC aggregation may
hung there. That is why dtx_37 is reported timeout in CI test.

Enable dtx_37.

Signed-off-by: Fan Yong <fan.yong@intel.com>

### CODE REVIEW

**Review by liuxuezhao — DISMISSED** [review]

**Comment by Nasf-Fan** [comment]
Refresh for merge conflict.

File: src/object/srv_ec_aggregate.c [hunk 1]
```diff
@@ -1645,7 +1646,7 @@ agg_process_stripe(struct ec_agg_entry *entry)
 
 	/* Query the parity */
 	rc = vos_iterate(&iter_param, VOS_ITER_RECX, false, &anchors,
-			 agg_recx_iter_pre_cb, NULL, entry, NULL);
+			 agg_recx_iter_pre_cb, NULL, entry, dth);
```
Comment (jgmoore-or) [inline src/object/srv_ec_aggregate.c:1649]: The return code for this (nested) invocation of vos_iterate does not currently propagate up to the outer iterator.  If this is necessary for your fix, then I can redo the return code handling.

Comment (Nasf-Fan) [reply src/object/srv_ec_aggregate.c:1649]: I refreshed the patch to resolve such issue.

**Review by jgmoore-or — APPROVED** [review]

**Comment by Nasf-Fan** [comment]
@wangdi1 , @liuxuezhao ,

would you please to help review the patch? Thanks!

**Review by liuxuezhao — APPROVED** [review]

**Comment by Nasf-Fan** [comment]
Another PR https://github.com/daos-stack/daos/pull/4563 verified the patch validity.

**Review by jolivier23 — APPROVED** [review]

**Pull request merged**